nix/default.nix

```diff
@@ -3,34 +3,31 @@
 , config ? { }
 , overlays ? [ ]
 , sourcesOverride ? { }
+, sources
+, isInFlake
+, haskellNix
 , checkMaterialization ? false
 , enableHaskellProfiling ? false
 }:
 let
-  sources = import ./sources.nix { inherit pkgs; }
-    // sourcesOverride;
-  haskellNix = import sources."haskell.nix" {
-    sourcesOverride = {
-      hackage = sources."hackage.nix";
-      stackage = sources."stackage.nix";
-    };
-  };
-
-  extraOverlays =
-    # Haskell.nix (https://github.com/input-output-hk/haskell.nix)
-    haskellNix.overlays
-    # our own overlays:
-    ++ [
+  ownOverlays =
+    [
       # Modifications to derivations from nixpkgs
       (import ./overlays/nixpkgs-overrides.nix)
       # fix r-modules
       (import ./overlays/r.nix)
     ];
 
+  # haskell-nix has to be used differently in flakes/no-flakes scenarios:
+  # - When imported from flakes, 'haskellNix.overlay' needs to be passed here.
+  # - When imported from default.nix without flakes, default to haskellNix.overlays
+  haskellNixOverlays = if isInFlake then [ haskellNix.overlay ] else haskellNix.overlays;
+
+  extraOverlays = haskellNixOverlays ++ ownOverlays;
+
   pkgs = import sources.nixpkgs {
-    inherit system crossSystem;
+    inherit config system crossSystem;
     overlays = extraOverlays ++ overlays;
-    config = haskellNix.config // config;
```

Comment:
This is important. I think we've now lost the `haskell.nix` config entirely?

Reply:
Hm, pretty sure I had mentioned that somewhere but i really don't recall when/where. I tried with and without the config and it didn't change the output:

```nix
{                                                                                                                                                                                                                                                                                                    
  # allow building for windows                                                                                                                                                                                                                                                                       
  allowUnsupportedSystem = true;                                                                                                                                                                                                                                                                     
  # we want the 64bit wine version                                                                                                                                                                                                                                                                   
  # See overlays/wine.nix as well!                                                                                                                                                                                                                                                                   
  wine.build = "wine64";                                                                                                                                                                                                                                                                             
  wine.release = "stable";                                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                     
  # This is marked insecure on 20.03, but many packages still depend on it                                                                                                                                                                                                                           
  permittedInsecurePackages = [ "openssl-1.0.2u" ];                                                                                                                                                                                                                                                  
}       
```

Reply:
That's true today, but what if they add important stuff in future? If they give us config stuff, we should use it, otherwise we're pointlessly diverging which sets us up for trouble later.

Reply:
Fixed in https://github.com/input-output-hk/plutus/pull/3221/commits/aeae158942f1670868c69713c06ebb940101e5cb